python/ray/data/tests/test_stats.py

```diff
@@ -308,7 +316,7 @@ def test_dataset__repr__(ray_start_regular_shared):
     assert canonicalize(repr(ds2._plan.stats().to_summary())) == (
         "DatasetStatsSummary(\n"
         "   dataset_uuid=U,\n"
-        "   base_name=MapBatches(<lambda>),\n"
+        "   base_name=DoRead->MapBatches,\n"
```

Comment:
What are the reasons for these stats changes? Are we fusing things we didn't before the optimizer flip?

Reply:
At least for this case, the fusion behavior is the same. It looks that previously, for a fused operator, the name doesn't display the full chain, but just the last op name.